tests/dao_tests/test_questionnaire_dao.py

```diff
@@ -205,3 +208,44 @@ def test_parsing_external_id(self):
         })
 
         self.assertEqual('FORM_1A', model.externalId)
+
+    def test_questionnaire_payload_cannot_create_new_codes(self):
```

Comment:
Suggestion only:  I wondered whether it's worth creating an API test as well that does a POST of a questionnaire_response payload that you expect to be rejected because it contains unrecognized codes that were not in the Questionnaire test setup?